Content.Server/Payload/EntitySystems/PayloadSystem.cs

```diff
@@ -0,0 +1,151 @@
+        foreach (var (name, data) in trigger.Components)
+        {
+            if (!_componentFactory.TryGetRegistration(name, out var registration))
+                continue;
+
+            if (HasComp(uid, registration.Type))
+                continue;
+
+            if (_componentFactory.GetComponent(registration.Type) is not Component component)
+                continue;
+
+            component.Owner = uid;
+
+            if (_serializationManager.Copy(data, component, null) is Component copied)
+                EntityManager.AddComponent(uid, copied);
+
+            trigger.GrantedComponents.Add(registration.Type);
+        }
+    }
+
+    private void OnEntityRemoved(EntityUid uid, PayloadCaseComponent component, EntRemovedFromContainerMessage args)
+    {
+        if (!TryComp(args.Entity, out PayloadTriggerComponent? trigger))
+            return;
+
+        trigger.Active = false;
+
+        foreach (var type in trigger.GrantedComponents)
+        {
+            EntityManager.RemoveComponent(uid, type);
+        }
+
+        trigger.GrantedComponents.Clear();
+    }
+
+    private void HandleChemichalPayloadTrigger(EntityUid uid, ChemicalPayloadComponent component, TriggerEvent args)
+    {
+        if (component.BeakerSlotA.Item is not EntityUid beakerA)
+            return;
+
+        if (component.BeakerSlotB.Item is not EntityUid beakerB)
+            return;
+
+        if (!TryComp(beakerA, out FitsInDispenserComponent? compA))
+            return;
+
+        if (!TryComp(beakerB, out FitsInDispenserComponent? compB))
+            return;
+
+        if (!_solutionSystem.TryGetSolution(beakerA, compA.Solution, out var solutionA))
+            return;
+
+        if (!_solutionSystem.TryGetSolution(beakerB, compB.Solution, out var solutionB))
+            return;
+
+        if (solutionA.TotalVolume == 0 || solutionB.TotalVolume == 0)
+            return;
+
+        var solStringA = SolutionContainerSystem.ToPrettyString(solutionA);
+        var solStringB = SolutionContainerSystem.ToPrettyString(solutionB);
+
+        _logSystem.Add(LogType.ChemicalReaction,
+            $"Chemical bomb payload {ToPrettyString(uid):payload} at {Transform(uid).MapPosition:location} is combining two solutions: {solStringA:solutionA} and {solStringB:solutionB}");
+
+        solutionA.MaxVolume += solutionB.MaxVolume;
+        _solutionSystem.TryAddSolution(beakerA, solutionA, solutionB);
+        solutionB.RemoveAllSolution();
+
+        // The grenade might be a dud. Redistribute solution:
+        var tmpSol = _solutionSystem.SplitSolution(beakerA, solutionA, solutionA.CurrentVolume * solutionB.MaxVolume / solutionA.MaxVolume);
+        _solutionSystem.TryAddSolution(beakerB, solutionB, tmpSol);
+        solutionA.MaxVolume -= solutionB.MaxVolume;
+        _solutionSystem.UpdateChemicals(beakerA, solutionA, false);
```

Comment:
Currently unless a chemnade is actually explosive /  deals damage, the beakers & solutions won't actually be destroyed. This means things like cleaning or smoke grenades are actually re-usable if you deconstruct them and re-fill the beakers.

I'm not 100% sure about this, but fixing this would require allowing the payload to determine whether the case entity should get deleted.

Reply:
I think this is fine.

Reply:
Tgstation has had reusable grenades for a couple months now. Two things about this.

1. You might want to make used chem grenades noticeably different from unused chem grenades. This prevents cases of people mistaking spent grenades from usable grenades. Alternatively leave them that way if it's funnier than it is inconvenient.
2. If this has logging on use you might want to add some sort of manually-resettable latch or something. Tg's didn't initially and a few admins crashed because of logging spam.

Reply:
1 - Yeah that would probably be useful. But it would also requires some not insignificant changes grenade sprite & logic, so that the payload type/state can determine the grenade visuals. This'd require splitting the grenade into separate layers (which it probably already should be) and then allowing different payloads/triggers to add their own layers to the grenade. For now I'm just gonna keep it as is and do that in a future PR.

2 - doesn't yet have a log on timer start, but will add one. the timer triggers should already be un-spammable.